docs/workflows.md

```diff
@@ -0,0 +1,114 @@
+# TorchServe Workflows
+
+TorchServe can be used for serving ensemble of models & functions (python) through Workflow APIs. 
+
+It utilizes [REST based APIs](rest_api.md) for workflow management and predictions.
+
+A Workflow is served on TorchServe using a workflow-archive(.war) which comprises of following: 
+
+## Workflow Specification file
+
+A workflow specification is a YAML file which provides the details of the models to be executed and a DAG for defining data flow.
```

Comment:
@dhanainme Please add some clarification on how the gpu assignment would be handled here.